etna/packages/magmaR/R/retrieve-special.R

```diff
@@ -82,16 +88,15 @@ retrieveIds <- function(
 #' @describeIn retrieve_SpecialCases Retrieve all the attribute options for a given project-model pair.
 #' @export
 retrieveAttributes <- function(
-    projectName = "ipi",
-    modelName = "patient",
+    projectName,
+    modelName,
     token = .get_TOKEN(),
     ...
 ) {
-
-    rec <- retrieveIds(projectName, modelName, token, ...)[1]
     
-    .retrieve(
-        projectName, modelName, recordNames = rec,
-        attributeNames = "all", format = "tsv",
-        names.only = TRUE, token = token, ...)
+    template <- retrieveTemplate(projectName, token, ...)
+    
+    atts <- names(template$models[[modelName]]$template$attributes)
+    
+    atts[!atts %in% c("created_at", "updated_at")]
```

Comment:
I'm not sure if it makes sense to hide (as this line does) versus show these???
These attributes are generally marked 'hidden' and thus don't get /retrieved when attributeNames = "all", nor are they displayed in timur/map, so hiding them makes sense to me.

Reply:
Can't test right now (🚿) but it looks like they are only marked `hidden` so they should show in a retrieve; the decision to show things called hidden is left to clients. If hidden is really censoring retrieve i might favor relaxing its stringency for these (which are super useful if a bit dodgy being phi-ish).

Reply:
The output of the deleted code (a retrieve with attributeNames = "all", then gave the rownames) was the same.  magma/retrieve does not give these attributes, at least not for `format = tsv`.

Reply:
Will probably revert this specific set of code changes anyway cuz looking at the fixtures reminded me that templates can actually be gigantic (because of matrix 'options' especially!) compared to the output of a `retrieve(..., recordNames = IDs[1], attributeNames = "all")`

Reply:
Yeah the matrix/template thing is dumb and unsustainable, we need a better way to distribute that information that is less cumbersome, I think it really slows down a lot of data retrieval that could be much faster to have these huge templates always being handed down.

Reply:
@graft I've been thinking one easy fix would to make an optional argument to _not_ return the template in payloads.  That would keep the system completely backwards compatible, but allow new clients to set this argument and avoid the slow painful realizations of the template on every load.

Reply:
pretty sure @graft and I have discussed that as a possibility at some point too!  As long as the default is `true` / `give the template`, shouldn't break any current code!